org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -78,10 +79,15 @@ public SourceAssistProcessor(PreferenceManager preferenceManager) {
 		IType type = getSelectionType(context);
 
 		// Organize Imports
-		TextEdit organizeImportsEdit = getOrganizeImportsProposal(context);
-		Optional<Either<Command, CodeAction>> organizeImports = convertToWorkspaceEditAction(params.getContext(), context.getCompilationUnit(), CorrectionMessages.ReorgCorrectionsSubProcessor_organizeimports_description,
-				CodeActionKind.SourceOrganizeImports, organizeImportsEdit);
-		addSourceActionCommand($, params.getContext(), organizeImports);
+		if (preferenceManager.getClientPreferences().isAdvancedOrganizeImportsSupported()) {

```

Comment:
The if conditional could be put insider getOrganizeImportsAction to make logic simple.

Reply:
This `if` is to fall back to the old organizeImport logic if the JDT LS client doesn't implement chooseImports command yet. Don't belong to getOrganizeImportsAction function.